File: persistence/src/main/java/io/split/client/thin/internal/persistence/AttributesEntity.java
package io.split.client.thin.internal.persistence;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "attributes")
public class AttributesEntity {
    @PrimaryKey
    @NonNull
    public final String key;
    @NonNull
    public final String json;
    public final long updatedAt;

    public AttributesEntity(@NonNull String key, @NonNull String json, long updatedAt) {
        this.key = key;
        this.json = json;
        this.updatedAt = updatedAt;
    }
}

File: persistence/src/main/java/io/split/client/thin/internal/persistence/AttributesEntity.kt
package io.split.client.thin.internal.persistence

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "attributes")
internal data class AttributesEntity(
    @PrimaryKey val keyHash: String,
    val attrHash: String,
    val changeNumber: Long,
    val lastUpdateTimestamp: Long? = null
)
